ml-algorithms/src/main/java/org/opensearch/ml/engine/tools/SearchAlertsTool.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.ml.engine.tools;
+
+import static org.opensearch.ml.common.utils.StringUtils.gson;
+
+import java.util.List;
+import java.util.Map;
+
+import org.opensearch.client.Client;
+import org.opensearch.client.node.NodeClient;
+import org.opensearch.commons.alerting.AlertingPluginInterface;
+import org.opensearch.commons.alerting.action.GetAlertsRequest;
+import org.opensearch.commons.alerting.action.GetAlertsResponse;
+import org.opensearch.commons.alerting.model.Table;
+import org.opensearch.core.action.ActionListener;
+import org.opensearch.ml.common.output.model.ModelTensors;
+import org.opensearch.ml.common.spi.tools.Parser;
+import org.opensearch.ml.common.spi.tools.Tool;
+import org.opensearch.ml.common.spi.tools.ToolAnnotation;
+
+import lombok.Getter;
+import lombok.Setter;
+
+@ToolAnnotation(SearchAlertsTool.NAME)
+public class SearchAlertsTool implements Tool {
+    public static final String NAME = "SearchAlertsTool";
+    private static final String DEFAULT_DESCRIPTION = "Use this tool to search alerts.";
+
+    @Setter
+    @Getter
+    private String name = SearchAlertsTool.NAME;
```

Comment:
I think from other tool renames (earlier today) this is now `TYPE`. I may be wrong, a rebase would determine that.

Reply:
Tool interface has both a name and type, from the changes, it looks like name and type are both set to the same value. I'll update.

Reply:
I think (@jngz-es please confirm) that `NAME` is a default name but can be changed with the name getters/setters, while `TYPE` is unchanging?

Reply:
Yes, correct.

Reply:
Cool thanks for clarifying. I've updated to be consistent with the recent changes to the other tools.